src/rai_core/rai/communication/ros2/api/action.py

```
@@ -0,0 +1,287 @@
+        callback_group: Optional[rclpy.node.CallbackGroup] = None,
+        goal_callback: Callable[[IROS2Message], GoalResponse] = default_goal_callback,
+        handle_accepted_callback: Callable[
+            [ServerGoalHandle], None
+        ] = default_handle_accepted_callback,
+        cancel_callback: Callable[
+            [IROS2Message], CancelResponse
+        ] = default_cancel_callback,
+        goal_service_qos_profile: QoSProfile = qos_profile_services_default,
+        result_service_qos_profile: QoSProfile = qos_profile_services_default,
+        cancel_service_qos_profile: QoSProfile = qos_profile_services_default,
+        feedback_pub_qos_profile: QoSProfile = QoSProfile(depth=10),
```

Ruff (0.8.2)

Do not perform function call `QoSProfile` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)